tools/GenerateCompilationDatabase.ps1

```diff
@@ -0,0 +1,166 @@
+
+foreach ($binlogFile in $binlogFiles) {
+    if (-Not (Test-Path $binlogFile)) {
+        Write-Error "Binlog file not found: $binlogFile"
+        exit 1
+    }
+
+    & $msBuildPath $binlogFile /v:normal /noconlog /flp:logfile=temp.log
+
+    Select-String -Path "temp.log" -Pattern "Target ""ClCompile"" in file","Cl.exe"  |
+        ForEach-Object { $_.Line } |
+        Where-Object { $_ -notmatch "Tracker.exe" } |
+        Out-File -FilePath "temp-filtered.log" -Append -Encoding utf8
+
+    Remove-Item "temp.log"
+
+    Write-Host "Processed binlog file: $binlogFile"
+}
+
+Write-Host "Filtered log file generated at: $(Get-Location)\temp-filtered.log"
+
+$contextPath = ""
+
+# for each line in temp-filtered.log, if it is a "Target " line, extract the project directory and save in the contextPath variable
+# if it is a "Cl.exe" line, prepend the contextPath to every source file in that line only if it is
+# a relative path, then output the modified line to temp-filtered-2.log
+# The source files are the arguments that ends with .cpp, .c, or .h
+# They can have spaces in them. If so, they are enclosed in quotes.
+
+$lines = Get-Content "temp-filtered.log"
+
+Write-Host "Processing filtered log to adjust file paths..."
+
+foreach ($line in $lines) {
+    if ($line -match 'Target "ClCompile" in file "([^"]+)" from project "([^"]+)"') {
+        $contextPath = Split-Path $matches[2] -parent
+        Write-Host "Context path set to: $contextPath"
+    } elseif ($line -match 'Cl\.exe (.+)$') {
+        $clLine = $matches[1]
+        $args = $clLine -split ' (?=(?:[^"]*"[^"]*")*[^"]*$)'
+
+        $modifiedArgs = @()
+        foreach ($arg in $args) {
+            if ($arg -match '^(.*\.(cpp|c|h))$' -or $arg -match '^"(.*\.(cpp|c|h))"$') {
+                $filePath = $arg.Trim('"')
+                if (-Not ([System.IO.Path]::IsPathRooted($filePath))) {
+                    $filePath = Join-Path $contextPath $filePath
+                }
+                if ($arg.StartsWith('"') -and $arg.EndsWith('"')) {
+                    $modifiedArgs += '"' + $filePath + '"'
+                } else {
+                    $modifiedArgs += $filePath
+                }
+            } else {
+                $modifiedArgs += $arg
+            }
+        }
+
+        $modifiedLine = "Cl.exe " + ($modifiedArgs -join ' ')
+        Add-Content -Path "temp-filtered2.log" -Value $modifiedLine
+    }
+}
+
+Write-Host "Adjusted file paths and generated: $(Get-Location)\temp-filtered2.log"
+
+$ms2ccPath = Join-Path $PSScriptRoot "ms2cc"
+$ms2ccExe = Join-Path $ms2ccPath "ms2cc.exe"
+
+if (-Not (Test-Path $ms2ccExe)) {
+    Write-Host "Downloading ms2cc..."
+    $ms2ccUrl = "https://github.com/freddiehaddad/ms2cc/releases/download/v$Ms2ccVersion/ms2cc-$Ms2ccVersion.zip"
+    $zipPath = Join-Path $PSScriptRoot "ms2cc-$Ms2ccVersion.zip"
+
+    try {
+        Invoke-WebRequest -Uri $ms2ccUrl -OutFile $zipPath -UseBasicParsing
```

Comment:
`DevCheck` uses `curl.exe` with -L (follow redirects) and -# (show progress)

How's Invoke-WebRequest compare?

Reply:
Invoke-WebRequest was the default I found for downloading something with powershell. Looking at the documentation, it does not seems to have the same capabilities as the ones you listed for curl.exe. Do you think it is worth changing it to use curl?